src/Temporalio/Client/ITemporalClient.cs

```diff
@@ -0,0 +1,14 @@
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Interface to a client to Temporal.
+    /// </summary>
+    /// <seealso cref="TemporalClient" />
+    public interface ITemporalClient : Worker.IWorkerClient
```

Comment:
In general I am not a fan of interfaces just for the sake of having them. But it's the .NET way for these things that may be DI-based, so I accommodate. That means that I will have to put all those `StartWorkflowAsync` overloads on here. One might think I could put one on here and make everything else an extension method, but .NET prefers putting things on the types themselves if you can, and extension methods if you can't (for compat or external-code reasons). See https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods#general-guidelines

Reply:
On this topic, recommend implementation classes (e.g. TemporalClient) are scoped to internal instead of public. A smaller public footprint means a smaller domain to learn for sdk consumers. E.g. when to use ITemporalClient vs. TemporalClient - easier to only see ITemporalClient & then the sdk can evolve the internals with less risk of breaking sdk api. Essentially interfaces and/or abstract classes & supporting types etc. would be the only public types an sdk consumer should have to see.